synthesizer/process/src/stack/execute.rs

```diff
@@ -355,6 +375,26 @@ impl<N: Network> StackExecute<N> for Stack<N> {
                     Operand::NetworkID => {
                         bail!("Illegal operation: cannot retrieve the network id in a function scope")
                     }
+                    // If the operand is the checksum, retrieve the checksum from the stack.
+                    Operand::Checksum(program_id) => {
+                        let checksum = match program_id {
```

Comment:
Are we allowing these operands in the function scope?

Reply:
We can consider removing, just to reduce the scope of the PR. It doesn't directly benefit program upgradability, and it doesn't seem consistent to allow `Edition` but not `NetworkID`.

Reply:
Yeah I think we want to allow them.
The use case I had in mind was:
- program passes either `Edition` or `Checksum` to finalize
- finalize checks that it matches the expected on-chain version
A developer could use this pattern to invalidate "old" transactions from users, providing a cleaner UX.
Obviously a malicious developer can still muck about.